### PR TITLE
Rename dss2! to dss!

### PR DESCRIPTION
**Pull request by sriharshakandala**
Rename dss2 functions to dss

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by sriharshakandala** [comment]
bors try

File: src/Spaces/dss_transform.jl [hunk 1]
```diff
@@ -196,6 +196,13 @@ function weighted_dss_start! end
 function weighted_dss_internal! end
 function weighted_dss_ghost! end
 
+# for backward compatibility
+function weighted_dss2! end
+function weighted_dss_start2! end
+function weighted_dss_internal2! end
+function weighted_dss_ghost2! end
+function dss2! end
```
Comment (simonbyrne) [inline src/Spaces/dss_transform.jl:204]: where are these methods defined?

Comment (sriharshakandala) [reply src/Spaces/dss_transform.jl:204]: These methods are defined in `src/Fields/Fields.jl`. These are just used for backward compatibility and will be eventually removed.

Comment (simonbyrne) [reply src/Spaces/dss_transform.jl:204]: But it looks like not all the same methods are defined as before?

Comment (sriharshakandala) [reply src/Spaces/dss_transform.jl:204]: Currently, only dss functions defined in the fields are currently called from external packages. So, backward compatible definitions are only added for these functions. The remaining functions are only used internally!

**Pull request merged**